Assets/Scripts/Enemy Folder/Enemy.cs

```diff
@@ -58,73 +89,85 @@ void Update()
         aggroTrigger.enabled = true;
-        isAttacking = false;
     }
 
-    IEnumerator Attack()
+    public Vector3 GetHome()
     {
-        canAttack = false;
-        isAttacking = true;
-
-        yield return new WaitForSeconds(3.0f);
-
-        if (targetUnit)
-        {  
-            float distanceToTarget = Vector3.Distance(transform.position, targetUnit.transform.position);
-
-            if (distanceToTarget < 5)
-            {
-                targetUnit.GetComponent<Player>().TakeDamage(1);
-            }
-        }
-        canAttack = true;
+        return home;
     }
-
-    public void TakeDamage(int damageValue)// Move to a seperate damage handler maybe
+    public GameObject GetTargetUnit()
     {
-        enemyDataInstance.CurrentHealth -= damageValue;
-        StartCoroutine(Hit());
-
-        if (enemyDataInstance.CurrentHealth <= 0)
-        {
-            Death();
-        }
+        return targetUnit;
+    }
+    public EnemyUnitData GetEnemyUnitData()
+    {
+        return enemyDataInstance;
     }
 
-    IEnumerator Hit()// To be replaced by animations
+    public async void Hit()// To be replaced by animations 
     {
         GetComponentInChildren<Renderer>().material = redMaterial;
-        yield return new WaitForSeconds(0.5f);
+        await new WaitForSeconds(0.5f);
         GetComponentInChildren<Renderer>().material = greenMaterial;
     }
 
-    void Death()
+    void Death(int id)
     {
-        GameObject clone = Instantiate(enemyDataInstance.Drop, transform.position, Quaternion.identity);
-        clone.GetComponent<Item>().SetData(enemyDataInstance.Data);
+        if (id != enemyDataInstance.UnitID) { return; }
+
+        GameObject clone = Instantiate(enemyDataInstance.DropObject, transform.position, Quaternion.identity);
+        clone.GetComponent<Item>().SetData(enemyDataInstance.DropData);
+
+        transform.GetComponent<NavMeshAgent>().enabled = false;
 
         Vector3 position = transform.position;
         position.y -= 40f;
         transform.position = position;
 
         Destroy(gameObject, 3.0f);
     }
+
+    public void DealDamage()
+    {
+        if (targetUnit)
+        {
+            float distanceToTarget = Vector3.Distance(transform.position, targetUnit.transform.position);
+
+            if (distanceToTarget <= enemyDataInstance.AttackRange)
+            {
+                DamageHandler.ApplyDamage(targetUnit.GetComponent<Player>(), enemyDataInstance.BasicAttackDamage);
+            }
+        }
+    }
+
+    public void ControlAnimations(MonsterStates animationName, bool isPlaying)
+    {
+        SerializedDictionary<MonsterStates, Animation> clips = enemyDataInstance.Animations;
+
+        Animation aniClip;
+        clips.TryGetValue(animationName, out aniClip);
+
+        if (aniClip == null) return;
+
+        if (isPlaying)
+        {
+            aniClip.Play();
+        }
+        else
+        {
+            aniClip.Stop();
+        }
+    }
```

Comment:
how about
```csharp
public void PlayAnimation(string index) {
    enemyDataInstance.Animations[index]?.Play();
}

public void Stop Animations(string index) {
    enemyDataInstance.Animations[index]?.Stop();
}
```

Reply:
like this?
`
public void ControlAnimations(MonsterStates animationName, bool isPlaying)
    {
        if (isPlaying)
        {
            enemyDataInstance.Animations[animationName]?.Play();
        }
        else
        {
            enemyDataInstance.Animations[animationName]?.Stop();
        }
    }
`